### DISCUSSION TITLE
给InternLM加入联网功能，搭建属于自己的联网助手（选题参考）（已满）

### DISCUSSION THREAD

**Discussion by sanbuphy** [InternLM/Tutorial#383] [project]
## 背景

在当今互联世界中，互联网的使用已经成为我们日常生活中不可或缺的一部分。从浏览网页到使用在线服务，互联网为我们提供了丰富的信息和资源。为了更好地满足您的需求，我们决定对InternLM / InternLM2这一强大的语言模型进行升级，加入联网功能，甚至帮助你实现一个自动化爬虫。
通过将联网功能整合到InternLM中，我们将为您提供以下优势和功能：

实时信息获取：通过联网功能，您可以随时随地获取最新的新闻、文章、数据等信息。您的联网助手将为您提供即时的搜索结果和相关资讯。

个性化推荐：基于您的兴趣和历史偏好，您的联网助手将学习并提供个性化的内容推荐。这将帮助您更快地找到您感兴趣的信息，并探索新的领域。

在线服务集成：联网助手将与各种在线服务进行集成，例如电子邮件、日历、社交媒体等。您可以通过联网助手访问和管理您的在线账户，提高工作和生活的便利性。

智能问答与解决方案：联网助手将能够利用互联网上的知识库和资源，提供更准确和全面的答案和解决方案。无论您遇到何种问题，联网助手都将努力为您提供帮助。

通过为InternLM加入联网功能，我们致力于提供一个全面且个性化的在线助手，以满足您在信息获取、任务管理和问题解决方面的需求。无论是在工作中还是在日常生活中，您的联网助手将成为您的忠实伙伴，为您提供帮助和支持。

## 技术栈

很简单，参考最新开源的强大知识领域助手——茴香豆的网络搜索部分即可：
https://github.com/InternLM/HuixiangDou/blob/main/huixiangdou/service/web_search.py

**Comment by Milkve** [comment]
想报名这个，但不清楚在线服务集成要如何实现

**Reply by sanbuphy** [reply]
这个式GPT写的，不用做都行；你可以让助教帮忙呼叫和拉队友（缺队长）

**Reply by zhaoxin111** [reply]
还缺人吗？

**Reply by sanbuphy** [reply]
你好，请让助教帮帮忙 艾特我介绍你 @zhaoxin111
